pycaw/tools/app_mixer/app_controll_and_gui.py

```diff
@@ -0,0 +1,248 @@
+import os
+
+# _____ ADVANCED WINDOWS COMMUNICATION _____
+from ctypes import pointer, windll
+
+# ____________ GUI WITH TKINTER ____________
+from tkinter import BOTTOM, HORIZONTAL, DoubleVar, StringVar, Tk
+from tkinter.font import Font
+from tkinter.ttk import (Button, Frame, Label, Scale, Scrollbar, Separator,
+                         Style)
+
+from comtypes import GUID
+
+# _________ GUI SCALE ON HIGH DPI _________
+windll.shcore.SetProcessDpiAwareness(2)
+# SetProcessDpiAwareness = 2: Per monitor DPI aware. 
+# shcore checks for the DPI when the app is
+# started and adjusts the scale factor whenever the DPI changes.
+# Normally tkinter apps are not automatically scaled.
+
+
+class RootFrame(Tk):
+    """Mixer Window"""
+    def __init__(self):
+        super().__init__()
+        # __________________ ROOT __________________
+        # create guid for volume changes from this mixer
+        self.lpcguid = pointer(GUID('{88e327ac-079a-49e5-8ac3-b2de61d3b0e5}'))
+
+        # __________ ROOT WINDOW MENU BAR __________
+        self.geometry("500x600")
+        self.title("Live Windows Audio Session Mixer")
+        # set icon:
+        dirname = os.path.dirname(__file__)
+        icon_path = os.path.join(dirname, 'app_mixer.ico')
+        self.iconbitmap(icon_path)
+
+        # ________________ STYLING ________________
+        s = Style()
+
+        s.configure("Active.TFrame", background='#007AD9')
+        # s.configure("Inactive.TFrame", background='#A2A2A2')
+        s.configure("Inactive.TFrame", background='#AF3118')
+
+        s.configure("Unmuted.TButton", foreground='#007AD9')
+        # s.configure("Muted.TButton", foreground='#C33C54')
+        s.configure("Muted.TButton", foreground='#AF3118')
+
+        # _________________ HEADER _________________
+        header = Frame(self)
+        title = Label(header, text="Live App Mixer",
+                      font=("Calibri", 20, "bold"))
+
+        # ________________ SEPARATE ________________
+        separate = Separator(header, orient=HORIZONTAL)
+
+        # ____________ ARRANGE ELEMENTS ____________
+        header.columnconfigure(0, weight=1)
+        title.grid(row=0, column=0, pady=10, sticky="W")
+        separate.grid(row=1, column=0, sticky="EW", pady=10)
+
+        # _______________ PACK FRAME _______________
+        header.pack(fill='x', padx=15)
+
+        # ____________ BUILD WITH PYCAW ____________
+        turbo_anonym = Label(self, text="created by TurboAnonym with pycaw",
+                             font=("Consolas", 10, "italic"))
+        turbo_anonym.pack(side=BOTTOM, fill="x", pady=6, padx=6)
+
+
+class AppRow(Frame):
+    """
+    Row for each app in mixer.
+    handles refreshing the gui if session is changed external.
+    handles user input and changing session volume/mute.
+    """
+
+    def __init__(self, RootFrameInstance, session, iid, session_manager):
```

Comment:
If `RootFrameInstance` is really an object instance rather than a type/class, then why isn't not snake case notation?

Reply:
... something learned again ;)